### ISSUE TITLE
Option extend-select not working

### ISSUE THREAD

**Issue by mikkuja** [MarketSquare/robotframework-robocop#1664]
I can't get linter [extend-select](https://robocop.dev/stable/configuration/configuration_reference/#extend-select) option to work. Am I doing something wrong or is this a bug?

For example when trying to extend rule `sleep-keyword-used` which is not enabled at default:

`test.robot`
```robotframework
*** Settings ***
Documentation    Foo


*** Test Cases ***
Foo
    [Documentation]    Foo
    Sleep    1
```

```bash
> robocop check --extend-select sleep-keyword-used test.robot
No issues found.

> robocop check --select sleep-keyword-used test.robot
test.robot:8:5 KW01 Sleep keyword with '1' sleep time found
```

Tested using latest release v7.2.0 and from latest commit eddfd967.

Also tested that same issue occurs when:
- Defining `extend-select` in `pyproject.toml` config file.
- With keyword `not-allowed-keyword`

Related to #1546

**Comment by bhirsz** [comment]
I found a suspect in the line:

  https://github.com/MarketSquare/robotframework-robocop/blob/main/src%2Frobocop%2Fruntime%2Fresolver.py#L64

Looks like a typo and it should not be patterns for rule name check. I will check it later but can you try using rule id?
